core/lib/libfdt/fdt_rw.c

```diff
@@ -285,7 +285,8 @@ int fdt_setprop(void *fdt, int nodeoffset, const char *name,
 	if (err)
 		return err;
 
-	memcpy(prop->data, val, len);
+	if (len)
+		memcpy(prop->data, val, len);
```

Comment:
It should be fine to call `memcpy()` with `len == 0` (C99 standard, 7.21.1 String function conventions, paragraph 2).  What's the problem? Some tool is complaining? What's the message? Can we fix the tool or the way you call it?